src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3329,6 +3329,18 @@ public override SyntaxNode BaseExpression()
         public override SyntaxNode TypedConstantExpression(TypedConstant value)
             => ExpressionGenerator.GenerateExpression(value);
 
+        private protected override SyntaxNode WithKeywordIndicatingExtensionMethod(SyntaxNode parameterDeclaration)
+        {
+            var parameter = (ParameterSyntax)parameterDeclaration;
+            return parameter.AddModifiers(SyntaxFactory.Token(SyntaxKind.ThisKeyword));
+        }
```

Comment:
if hte parameter has other modifiers already (like ref) does this do the right thing wrt ordering of modifiers?